activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -59,20 +59,23 @@ def new_time(year, mon, mday, hour, min, sec, microsec, offset = nil)
             end
           end
 
-          ISO_DATETIME = /\A(\d{4})-(\d\d)-(\d\d) (\d\d):(\d\d):(\d\d)(\.\d+)?\z/
+          ISO_DATETIME = /\A(\d{4})-(\d\d)-(\d\d) (\d\d):(\d\d):(\d\d)(?:\.(\d+))?\z/
```

Comment:
This would let us avoid the `usec >= 6` branch, wouldn't it?

```suggestion
          ISO_DATETIME = /\A(\d{4})-(\d\d)-(\d\d) (\d\d):(\d\d):(\d\d)(?:\.(\d{1,6})\d*)?\z/
```

Reply:
👍  Updated.